src/phpDocumentor/FilesystemFactory.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor;
+
+use \LogicException;
+use League\Flysystem\Adapter\Local;
+use League\Flysystem\Filesystem;
+use League\Flysystem\MountManager;
+use \DI\Container;
+
+/**
+ * Class FilesystemFactory
+ */
+final class FilesystemFactory
+{
+    /** @var MountManager */
+    private $mountManager;
+
+    /** @var Container */
+    private $container;
+
+    /** @var Filesystem */
+    private $filesystem;
+
+    /**
+     * @param MountManager $mountManager
+     * @param Container $container
+     */
+    public function __construct(MountManager $mountManager, Container $container)
+    {
+        $this->mountManager = $mountManager;
+        $this->container = $container;
+    }
+
+    /**
+     * Returns a Filesystem instance based on the scheme of the provided Dsn
+     *
+     * @param Dsn $dsn
+     * @return Filesystem
+     */
+    public function create(Dsn $dsn)
+    {
+        $dsnId = spl_object_hash($dsn);
+
+        if (! $this->setFilesystemWhenCached($dsnId)) {
+            if ($dsn->getScheme() === 'file') {
+
+                $path = $dsn->getPath();
+                $this->filesystem = new Filesystem(new Local(__DIR__. "/" . $path));
```

Comment:
The path determination can cause issues: the $path may contain an absolute path (i.e. `file:///home/mvriel/Projects/phpDocumentor`) and if that happens it will be prefixed with the directory of this FilesystemFactory (because `__DIR__` does not return the current working directory, `getcwd` does).

Please check if the Local adapter supports receiving relative paths because you might not need to have to prefix the path.